lib/repl.js

```diff
@@ -284,14 +284,16 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
-
+    const pstrace = Error.prepareStackTrace;
+    Error.prepareStackTrace = prepareStackTrace(pstrace);
     internalUtil.decorateErrorStack(e);
+    Error.prepareStackTrace = pstrace;
     const isError = internalUtil.isError(e);
     if (e instanceof SyntaxError && e.stack) {
       // remove repl:line-number and stack trace
       e.stack = e.stack
-                 .replace(/^repl:\d+\r?\n/, '')
-                 .replace(/^\s+at\s.*\n?/gm, '');
+        .replace(/^repl:\d+\r?\n/, '')
+        .replace(/^\s+at\s.*\n?/gm, '');
```

Comment:
Is this still necessary with the `prepareStackTrace` function in place? And if yes, cant we just move that part in the prepareStackTrace function as well?

Reply:
I can poke around on it more - the change in formatting is because that's what I was doing and missed noticing that on my local diff.  It's tricky because of the way `Error.prepareStackTrace()` works. It's only ever called once for a given error, and the return value is by default (and typically in custom impls as well) a string. Because of the way errors are handled in the repl and domain, in the case of a `SyntaxError` specifically, it's a string before we have an opportunity to filter the stack frames in `prepareStackTrace()`. So I left it rather than rework more of the overall error handling.

Reply:
@BridgeAR I spent the morning looking into removing this and dealing with it in `Error.prepareStackTrace()` but the ramifications were wide reaching, especially with regard to how we handle `SyntaxError` as a `Recoverable` object so that defaulting into editor mode on multi-line input works as expected. I do think there could be some consolidation and overall improvement of the error handling, but I don't think this is the PR to do that in.